### DISCUSSION TITLE
Expanding the Bank - Deposits

### DISCUSSION THREAD

**Discussion by TechnicalSnark** [endless-sky/endless-sky#10764] [Ideas]
After several playthroughs and paying off the initial ship mortgages, it dawned on me that the bank stops being useful for players actively avoiding taking additional loans. I pondered, "What if the bank in Endless Sky operated similarly to banks on Earth?" Measures such as locking in deposited funds (similarly to I and EE bonds) and penalties for early withdrawal (similarly to certificates of deposit and money market account), coupled with low interest rates, would discourage abuse of deposit functionality.

Before I dive too deeply into the source code, though, I wanted to see if such functionality already exists. After checking out a handful of plugin websites, I didn't find an existing plugin that offers similar functionality.

Please advise.

**Comment by tibetiroka** [comment]
What would the benefit of a deposit be compared to investing the money in another ship?

**Reply by TechnicalSnark** [reply]
This may be anecdotal, but a significant number of my playthroughs have involved pirates spawning on top of my flagship or on top of one of my escort ships, which sometimes leads to that ship being destroyed. As an alternate to save-scumming, having a rainy-day/emergency fund on which to fall back could be beneficial, especially to players performing a permadeath/no-save-scumming playthrough. Also, the bank's offerings could be expanded to include high-risk, high-reward investments.

TL;DR: Immersion.

**Reply by TheSpacepirate25** [reply]
Maybe if you're in Navy space, if you die a certain amount could be taken out of your emergency fund to come and rescue you. In EVE Online, when your ship is destroyed you eject in a capsule that has no weapons. This could be a similar idea to EVE's capsule system.

**Reply by TechnicalSnark** [reply]
I was thinking more along the lines of "After buying ships, outfits, and goods, I have credits left over, so I'll let them sit in the bank and gain interest while I warp/scram/jump across the universe."

I personally don't mind save-scumming. I was just thinking, "What if the bank did more than just mortgages?"

I downloaded the source code, but my C++ is rustier than a century-old butterknife, so it will take me a while to figure out a way to implement a basic savings account.

**Comment by esksp** [comment]
> the bank stops being useful for players actively avoiding taking additional loans

Indeed, I have wondered too what the bank is useful for after you have paid back your initial loan.
Useful additional loans are hard to get, and it isn't too hard to make money in ES: Unless you're _really_ impatient, you can buy almost everything after jobbing for some time, especially in the high-rewards systems where sometimes a single quick transport mission can earn you over a million!
Also money is immaterial, you can easily carry millions on you, currency is the same all over the universe, so a bank doesn't really seem to have any use.

Let's analyze this: Why do we need banks in Real Life? 
1. First and foremost because it's _unsafe_ and _complicated_ to carry large amounts of cash around
2. Because there are _remote payments_ you can only make through some financial system
3. Because depositing money in a _savings account_ can earn you some additional profit
4. They can _change money_, because of different currencies

None of those apply to ES:
- Your money is immaterial and thus safe and has no weight/volume. Carrying millions is as easy as carrying a couple hundred credits.
- Payments are only done in the three shops (Trading, Shipyard and Outfitter), and those are easy to pay with your virtual money stash.
- There are no saving accounts, and given income isn't a problem in ES, they wouldn't really make any sense either: Life is cheap, jobs are plentiful, and you really need to have a lot of bad luck to come into financial difficulties.
- As for money change, there is no need for that. All systems, even those who haven't yet seen a human, gladly accept human currency at a 1:1 value (and apparently have a direct wire back to human banks, since they know your credit rating!...).

After paying back my loan, I haven't found any reason to click on it, it's just wasted GUI space...
I would indeed vote to drop the bank altogether. It's only use in the game is to limit the player's initial income, until (s)he has paid back the initial loan, then it's kind of pointless, and I don't see how this can be improved, unless you implement a complicated financial system with different currencies, add robbers, and give money a volume and weight. Sorry...

**Comment by TheSpacepirate25** [comment]
Really the only way to give the bank an actual use is to make new game mechanics. Maybe a gamerule that when you die, instead of reverting to your last save, you lose 30% of your bank balance and you're teleported to the nearest space station. But I agree, the bank is totally useless.
